reactfront/src/components/graph-utils.js

```diff
@@ -24,16 +24,14 @@ function roundToSigFigs(x, sigFigs) {
 
 // Formatted label including SI unit prefixes
 export function formatUnit(val, unit, sigFigs = 3) {
-  return Math.abs(val) >= 1_000_000_000_000  ? `${roundToSigFigs(val / 1_000_000_000_000, sigFigs)} T${unit}`
-       : Math.abs(val) >= 1_000_000_000      ? `${roundToSigFigs(val / 1_000_000_000,     sigFigs)} G${unit}`
-       : Math.abs(val) >= 1_000_000          ? `${roundToSigFigs(val / 1_000_000,         sigFigs)} M${unit}`
-       : Math.abs(val) >= 1_000              ? `${roundToSigFigs(val / 1_000,             sigFigs)} k${unit}`
-       : Math.abs(val) >= 1                  ? `${roundToSigFigs(val,                     sigFigs)} ${unit}`
-       : Math.abs(val) >= 0.001              ? `${roundToSigFigs(val * 1_000,             sigFigs)} m${unit}`
-       : Math.abs(val) >= 0.000_001          ? `${roundToSigFigs(val * 1_000_000,         sigFigs)} μ${unit}`
-       : Math.abs(val) >= 0.000_000_001      ? `${roundToSigFigs(val * 1_000_000_000,     sigFigs)} n${unit}`
-       : Math.abs(val) >= 0.000_000_000_001  ? `${roundToSigFigs(val * 1_000_000_000_000, sigFigs)} p${unit}`
-       : `0 ${unit}`
+  let prefixes = [["T",4], ["G",3], ["M",2], ["K",1], ["",0], ["m",-1], ["μ",-2], ["n",-3]]
+  for (let i = 0; i < prefixes.length; i++) {
+      let divider = Math.pow(1000, prefixes[i][1])
+      if (Math.abs(val) >= divider) {
+          return roundToSigFigs(val / divider, sigFigs) + ' ' + prefixes[i][0] + unit
+      }
+  }
+  return roundToSigFigs(val / Math.pow(1000, -4), sigFigs) + ' p' + unit
```

Comment:
Just one issue with the new version: `formatUnit(0, "V")` returns `0 pV` instead of `0 V`.

Reply:
Fixed now. Rip, can't believe I didn't see that